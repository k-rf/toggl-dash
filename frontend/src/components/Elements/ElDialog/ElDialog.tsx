import { Dialog, DialogProps } from "@mui/material";

type Props = Pick<DialogProps, "children" | "onClose" | "open" | "disableEscapeKeyDown" | "sx">;

export const ElDialog = (props: Props) => {
  return <Dialog {...props} />;
};

export type { Props as ElDialogProps };
