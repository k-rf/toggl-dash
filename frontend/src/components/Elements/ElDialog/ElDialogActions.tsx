import { DialogActions, DialogActionsProps } from "@mui/material";

type Props = Pick<DialogActionsProps, "children" | "sx">;

export const ElDialogActions = (props: Props) => {
  return <DialogActions {...props} />;
};
