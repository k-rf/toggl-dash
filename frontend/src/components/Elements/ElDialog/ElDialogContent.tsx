import { DialogContent, DialogContentProps } from "@mui/material";

type Props = Pick<DialogContentProps, "children" | "sx">;

export const ElDialogContent = (props: Props) => {
  return <DialogContent {...props} />;
};
