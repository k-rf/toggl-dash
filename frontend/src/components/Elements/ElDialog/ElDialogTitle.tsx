import { DialogTitle, DialogTitleProps } from "@mui/material";

type Props = Pick<DialogTitleProps, "children" | "sx">;

export const ElDialogTitle = (props: Props) => {
  return <DialogTitle {...props} />;
};
