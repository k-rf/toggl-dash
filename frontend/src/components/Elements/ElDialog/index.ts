export * from "./ElDialog";
export * from "./ElDialogActions";
export * from "./ElDialogContent";
export * from "./ElDialogTitle";
