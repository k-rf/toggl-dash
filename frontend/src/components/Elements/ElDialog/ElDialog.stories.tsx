import type { Meta, StoryObj } from "@storybook/react";

import { ElDialog } from "./ElDialog";
import { ElDialogActions } from "./ElDialogActions";
import { ElDialogContent } from "./ElDialogContent";
import { ElDialogTitle } from "./ElDialogTitle";

export default { component: ElDialog } as Meta<typeof ElDialog>;

type Story = StoryObj<typeof ElDialog>;

export const Default: Story = {
  args: {
    open: true,
    children: (
      <>
        <ElDialogTitle>Dialog Title</ElDialogTitle>
        <ElDialogContent>Dialog Content</ElDialogContent>
        <ElDialogActions>Dialog Actions</ElDialogActions>
      </>
    ),
  },
};
