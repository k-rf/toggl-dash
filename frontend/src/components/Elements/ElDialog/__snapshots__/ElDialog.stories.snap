// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Elements/ElDialog/ElDialog > Default 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r2:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1rd8hd1-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r2:"
        >
          Dialog Title
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          Dialog Content
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          Dialog Actions
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
